### PR TITLE
Better describe the equivalence triangulation=mesh=grid.

### LINKED ISSUE #16938: Use "mesh" instead of "grid" in step-1.
In various places in the documentation, we explain that "triangulation" = "mesh". In step-1, we use a third term, namely "grid" when outputting data into files and in screen messages. I suspect that this is due to the naming of `namespace GridGenerator`, but it is nonetheless unfortunate to use yet another term.

My vote would be to rename output files to `mesh-1/2.svg` and adjust the screen message accordingly. Opinions?

### PR DESCRIPTION
**Pull request by bangerth**
Fixes #16938.

### CODE REVIEW

File: examples/step-1/doc/intro.dox [hunk 1]
```diff
@@ -137,22 +137,47 @@ later on. Additionally, as of deal.II version 9.0, GridGenerator::hyper_shell()
...
-will have to have somewhere: An object of type Triangulation for the
-mesh; a call to the GridGenerator functions to generate a mesh; and
-loops over all cells that involve iterators (iterators are a
-generalization of pointers and are frequently used in the C++ standard
-library; in the context of deal.II, the @ref Iterators module talks
-about them).
+Let's come back to step-1, the current program. The goal of this
+program is to introduce you to the Triangulation class that is at the
+core of every finite element program. The name "triangulation" in this
+context is mostly historical: To finite element practitioner, the
```
Comment (kronbichler) [inline examples/step-1/doc/intro.dox:146]:
```suggestion
context is mostly historical: To finite element practitioners, the
```

File: examples/step-1/doc/intro.dox [hunk 2]
```diff
@@ -137,22 +137,47 @@ later on. Additionally, as of deal.II version 9.0, GridGenerator::hyper_shell()
...
+terms "triangulation", "mesh", and "grid" are all synonymous and
+describe a subdivision of a domain on which a differential equation is
+posed into cells of some kind. If the domain is two-dimensional, these
+cells may indeed be triangles, but they could also be quadrilaterals
+(four-sided objects such as squares and rectangles, and their
+deformations). In one space dimension, the cells are line segments. In
+three space dimensions, they can be tetrahedra, hexahedra (deformed
+cubes), pyramids (a four-sided base with three triangles connecting to
+a point at the top), and "wedges" (two triangles at the bottom and
+top, connected by three quadrilaterals). Collections of any such cells are
```
Comment (kronbichler) [inline examples/step-1/doc/intro.dox:156]: Do you want to state here that some part of the literature also refers to wegdes as "prisms"? For example, the wikipedia site uses the term "triangular prism", https://en.wikipedia.org/wiki/Types_of_mesh - but I have seen this name in some software packages like https://github.com/DLR-AMR/t8code and a number of scientific presentations, too.

File: examples/step-1/doc/intro.dox [hunk 3]
```diff
@@ -137,22 +137,47 @@ later on. Additionally, as of deal.II version 9.0, GridGenerator::hyper_shell()
...
+"triangulations" in common usage of the word even though they may not
+actually have triangles in them. All of them are also "grids" in
+common usage of the word even though the vertices are in neat rows
+parallel to the coordinate axes. In other words, whenever you read any
```
Comment (kronbichler) [inline examples/step-1/doc/intro.dox:160]: Here I do not entirely follow the text. Do you want to say that the classical definition of a grid has the vertices "neat rows parallel to the coordinate axes", but the term is also used for general point clouds? I think one understands what you would like to express, it's just that I don't quite follow with the logic of the text here.

File: examples/step-1/doc/intro.dox [hunk 4]
```diff
@@ -137,22 +137,47 @@ later on. Additionally, as of deal.II version 9.0, GridGenerator::hyper_shell()
...
+the key operation on containers is that one can *iterate* over its
+elements, and that's exactly what we will do below.
+
+Specifically, in the program we create two grids, one which is a
+regularly refined square (not very exciting, but a common starting
+grid for many problems), and one that is a more geometric attempt: a
+ring-shaped domain that is refined towards the inner edge. The process
+of refining the mesh in this way will illustrate how we iterate (i.e.,
+loop) over the elements of the triangulation (i.e., the cells of the
+mesh). You will see many many more such loops throughout the remainder
```
Comment (kronbichler) [inline examples/step-1/doc/intro.dox:177]: I would drop the second "many", but I realize it is personal taste, so feel free to decide what you like best:
```suggestion
mesh). You will see many more such loops throughout the remainder
```

File: include/deal.II/grid/tria.h [hunk 5]
```diff
@@ -550,12 +550,12 @@ namespace internal
  * associated with curves in 2d or surfaces in 3d) are the ones one wants to
  * use in the boundary element method.
  *
- * The name of the class is mostly hierarchical and is not meant to imply that
+ * The name of the class is mostly historical and is not meant to imply that
```
Comment (kronbichler) [inline include/deal.II/grid/tria.h:553]: That was an interesting statement before the change :rofl:

**Review by kronbichler — APPROVED** [review]
Nice, I like this. I have a few minor remarks.

File: examples/step-1/doc/intro.dox (hunk 2 above)
Comment (bangerth) [reply examples/step-1/doc/intro.dox:156]: Yes, good idea. I don't know why we called it "wedges", I think "prism" is the more commonly used term. But both are fine.

File: examples/step-1/doc/intro.dox (hunk 3 above)
Comment (bangerth) [reply examples/step-1/doc/intro.dox:160]: Yes, I think of the term "grid" as in Karopapier. I added another half sentence here, and mentioned the term "structured grid".

File: include/deal.II/grid/tria.h (hunk 5 above)
Comment (bangerth) [reply include/deal.II/grid/tria.h:553]: Yes. The longer I'm in the US, I find that I make the same mistakes as native speakers in that I accidentally write words that sounds similar but are different (in particular there/they're/their, but this is probably another example of the same problem).

This was introduced in #10979. Both @tjhei and @masterleinad missed it too :-)

**Comment by bangerth** [comment]
Fixed as requested.

**Review by kronbichler — APPROVED** [review]

**Pull request merged**